Fixing build with gcc 4.5

http://bugs.gentoo.org/show_bug.cgi?id=318429

Patch written by Kacper Kowalik <xarthisius.kk@gmail.com>

--- a/Synopsis/Parsers/Cxx/Types.cc
+++ b/Synopsis/Parsers/Cxx/Types.cc
@@ -169,7 +169,7 @@
 // Class Types::FuncPtr
 //
 
-FuncPtr::FuncPtr(Type::Type* ret, const Mods& premods, const Type::vector& params)
+FuncPtr::FuncPtr(Type* ret, const Mods& premods, const Type::vector& params)
         : m_return(ret), m_premod(premods), m_params(params)
 {}
 
